torch/_inductor/lowering.py

```diff
@@ -4545,7 +4545,7 @@ def reduce_any(x, dim=None, keepdim=False):
     return make_reduction("any")(x, axis=dim, keepdims=keepdim)
 
 
-@register_lowering(aten.max)
+@register_lowering(aten.max, type_promotion_kind=None)
```

Comment:
What was the type promotion that was happening previously ? It looks like only the tensor inputs should be used in type promotion: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/lowering.py#L181. So i'm curious exactly what was happening here

Reply:
Yes, only tensors are candidates for promotion, but when considering the dtype to promote to, `dim` gets considered as well since it is a Number: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/lowering.py#L187-L190, and the resulting dtype is an int64.

Presumably this is done because some number parameters are meant to be added or multiplied with the tensor